theano/tensor/basic.py

```diff
@@ -1427,21 +1427,31 @@ class Argmax(Op):
     nin = 2  # tensor, axis
     nout = 1
     E_axis = 'invalid axis'
-    __props__ = ()
+    __props__ = ('axis',)
```

Comment:
why this change? This break compatibility. It will be hard to fix, as we can't fix it with just a set_state method as we need to change all apply nodes that reuse that op, not just the op instance.

I would prefer that we don't change that. You can easily do get_scalar_constant_value(node.inputs[1]) to get the constant axis.